GTHC/web
# Schedule Automation

## Description

The feature to automatically create shifts based on user shifts, utilizing the Olson algorithm (check the [gthc gem](https://github.com/gthc/gthc)).

## Motivation and Context


One of our biggest, new features of this year. Helps with one of the most difficult tasks of tenting, going through people's availability and creating shifts for them.

## How Has This Been Tested?



1.  Run `yarn flush` to generate availabilities for fake generated users.
    a. Join a team with passcode "abc123" and create your own shifts.
2. Go to Calendar
3. Click the "Generate Shifts" button on the calendar, choose your settings (date range, type of tenting, and clearing other shifts) and see the shifts being generated.

## Screenshots (if appropriate):
![image](https://user-images.githubusercontent.com/13587407/71369117-c4b8c200-2577-11ea-911b-38fbe6ad09ee.png)

![image](https://user-images.githubusercontent.com/13587407/71369133-d5693800-2577-11ea-8815-b3c660cab3cf.png)


## Types of changes


## Checklist:
